### ISSUE TITLE
Scheduled Communications

### ISSUE THREAD

**Issue by StephenOTT** [DigitalState/Platform-Communication-Bundle#11]
Ability to Schedule a communication for sending as a specific date.

Few scenarios:

1. Generate messages right away so that messages are delivered to recipients at the Scheduled Time
1. Generate messages at the scheduled time, so that messages are delivered any time after scheduled time.

@sadortun Please provide some insights when you get a chance, as you had some good text for these scenarios.

**Comment by sadortun** [comment]
I think this should be integrated with an Oro Workflow.   When pressing SEND, the workflow can ask for the delivery time ( Now , or later) and then set the Communication in SENDING state or PENDING

**Comment by StephenOTT** [comment]
@sadortun where do you see the future date being stored?  A cron cycle looking for delivery dates that have passed?

**Comment by sadortun** [comment]
You can store the `sender At` date inside the Communication.  As you mentioned,The cron job that process the message awaiting to be sent can also check for this field to know when to start the transmission (KISS)

**Comment by StephenOTT** [comment]
Agreed.  Was also looking at the Oro WorkflowBundle "Processes" as a possibility in tandem with cron

**Comment by sadortun** [comment]
@StephenOTT  Got it working with the ORO Workflows

![test](https://cloud.githubusercontent.com/assets/794449/24040048/c13d3230-0add-11e7-8efe-3cc181ffea54.gif)
